lib/chef/knife/oci_helper.rb

```diff
@@ -1,43 +1,52 @@
 # Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved.
 
 require 'chef/knife'
-require 'knife-bmcs/version'
+require 'knife-oci/version'
 
 # rubocop:disable Metrics/ModuleLength
 class Chef
   class Knife
-    # BMCS helper module
-    module BmcsHelper
-      def bmcs_config
-        unless @bmcs_config
-          # Load config and profile first from command line args if available, then from knife.rb, then use the default.
-          config_file = config[:bmcs_config_file] || Chef::Config[:knife][:bmcs_config_file] || OracleBMC::ConfigFileLoader::DEFAULT_CONFIG_FILE
-          profile = config[:bmcs_profile] || Chef::Config[:knife][:bmcs_profile] || OracleBMC::ConfigFileLoader::DEFAULT_PROFILE
-          @bmcs_config = OracleBMC::ConfigFileLoader.load_config(config_file_location: config_file, profile_name: profile)
-          @bmcs_config.region = config[:region] if config[:region]
-
-          @bmcs_config.additional_user_agent = "Oracle-ChefKnifeOCI/#{::Knife::BMCS::VERSION}"
+    # OCI helper module
+    module OciHelper
+      def oci_config
+        unless @oci_config
+          @oci_config = OCI::ConfigFileLoader.load_config(config_file_location: config_file_location, profile_name: config_file_profile)
+          @oci_config.region = config[:region] if config[:region]
+
+          @oci_config.additional_user_agent = "Oracle-ChefKnifeOCI/#{::Knife::OCI::VERSION}"
         end
 
-        @bmcs_config
+        @oci_config
+      end
+
+      def config_file_location
+        # Load first from command line args if available, then from knife.rb, then use the default.
+        # For backwards compatibility with knife-bmcs, if oci version is not found in knife.rb, then check for bmcs version.
+        config[:oci_config_file] || Chef::Config[:knife][:oci_config_file] || Chef::Config[:knife][:bmcs_config_file] || OCI::ConfigFileLoader::DEFAULT_CONFIG_FILE
```

Comment:
since OCI::ConfigFileLoader::DEFAULT_CONFIG_FILE points ~/.oci/config, if this does not exist, do you have the logic to check ~/.oraclebmc/config?

Reply:
Good question. I've verified that this is working correctly. It turns out that the SDK's load_config method is not checking whether or not the config_file_location is set, but instead whether or not it equals the default value. So, that means that as long as OCI::ConfigFileLoader::DEFAULT_CONFIG_FILE is passed in, it will use the fallback logic if needed.

Reply:
oh, you are right.